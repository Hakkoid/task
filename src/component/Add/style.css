.Add{
	width: 300px;
	background: #eee;
	padding: 5px;
	height: 40px;
}

.Add__Input{
	height: 100%;
	width: 230px;
	border: none;
	padding: 5px;
}

.Add__Button{
	font-size: 15px;
	height: 100%;
	width: 50px;
	background: #fff;
	border: none;
	float: right;
	color: #439fc2;
	font-weight: 700;
}