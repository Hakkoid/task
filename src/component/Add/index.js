import React, { Component } from 'react';
import './style.css';
import PropTypes from 'prop-types'


class Add extends Component {
  constructor(props){
    super(props);
    this.state = { input: ""};
    this.handleChange = this.handleChange.bind(this);
    this.handleAdd = this.handleAdd.bind(this);
  }

  handleChange(e){
    this.setState({input: e.target.value})
  }

  handleAdd(){
    this.props.onAdd(this.state.input)
  }

  render() {
    return (
      <form className = 'Add' onSubmit = {(e) => { e.preventDefault(); this.handleAdd()}}>
        <input className = 'Add__Input' onChange = {this.handleChange} type="text" value = {this.state.input}/>
        <input type = 'submit' className = 'Add__Button' value = 'Add' />
      </form>
    );
  }
}

Add.propTypes = {
  onAdd: PropTypes.func.isRequired
}

export default Add
