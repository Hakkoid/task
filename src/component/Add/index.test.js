import React from 'react';
import { mount } from 'enzyme';
import Add from './index';
import renderer from 'react-test-renderer';


describe( 'Add', () => {

    it('renders without crashing', () => {
        mount(<Add onAdd={()=>{}}/>)
    })


    describe( 'onAdd event', () => {
        
        it('Click on the button should send the input state to the callback func', () => {

            let handler = (event) => { expect(event).toBe('one')}

            let component = mount(<Add onAdd = {handler}/>);
            component.setState({input: 'one'});
            component.find('.Add__Button').simulate('submit')
        })

        it('Submit form handler should send the input state to the callback func', () => {

            let handler = (event) => { expect(event).toBe('six')}

            let component = mount(<Add onAdd = {handler}/>);
            component.setState({input: 'six'});
            component.find('.Add').simulate('submit')
        })
    })


    it( 'Snapshot', () => {
        const component = renderer.create(
        <Add onAdd = { () => {}} />,
        );

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
})