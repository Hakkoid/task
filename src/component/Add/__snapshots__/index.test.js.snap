// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Snapshot 1`] = `
<form
  className="Add"
  onSubmit={[Function]}
>
  <input
    className="Add__Input"
    onChange={[Function]}
    type="text"
    value=""
  />
  <input
    className="Add__Button"
    type="submit"
    value="Add"
  />
</form>
`;
